pkg/edvprovider/couchdbedvprovider/couchdbedvprovider.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package couchdbedvprovider
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/trustbloc/edge-core/pkg/storage"
+	couchdbstore "github.com/trustbloc/edge-core/pkg/storage/couchdb"
+
+	"github.com/trustbloc/edv/pkg/edvprovider"
+	"github.com/trustbloc/edv/pkg/restapi/edv/edverrors"
+	"github.com/trustbloc/edv/pkg/restapi/edv/models"
+)
+
+const indexedField = "IndexName"
+
+// ErrMissingDatabaseURL is returned when an attempt is made to instantiate a new CouchDBEDVProvider with a blank URL.
+var ErrMissingDatabaseURL = errors.New("couchDB database URL not set")
+
+type couchDBIndexMappingDocument struct {
+	IndexName              string `json:"IndexName"`
+	MatchingEncryptedDocID string `json:"MatchingEncryptedDocID"`
+}
+
+// CouchDBEDVProvider represents a provider with functionality needed for EDV data storage.
+// It extends the edge-core storage provider interface with additional functionality that's needed for EDV operations.
+type CouchDBEDVProvider struct {
+	*couchdbstore.Provider
+}
+
+// NewProvider instantiates Provider
+func NewProvider(databaseURL string) (edvprovider.EDVProvider, error) {
+	var edvProv *CouchDBEDVProvider
+
+	couchDBProvider, err := couchdbstore.NewProvider(databaseURL)
+	if err != nil {
+		if err.Error() == "hostURL for new CouchDB provider can't be blank" {
+			return edvProv, ErrMissingDatabaseURL
+		}
+
+		return edvProv, err
+	}
+
+	return &CouchDBEDVProvider{Provider: couchDBProvider}, nil
+}
+
+// CreateIndexOnSingleField creates an index on a single field.
+// Since CouchDB won't let us index a field within an array (which is where encrypted indices are defined),
```

Comment:
I don't think we should air dirty laundry in the comments, i.e. the comment should describe what the function is doing. The comment in the PR (or issue) may contain the reasons for doing something a certain way.

Reply:
Done.